### DISCUSSION TITLE
Attempting to connect to the WebSocket server with the same IP when its already connected.

### DISCUSSION THREAD

**Discussion by MichaelEpicA** [jchristn/WatsonWebsocket#137] [General]
It just fails to connect, is there any setting that might be causing this or anything? I just want to be able to connect with the same IP, as it takes a bit to close the connection, so it won't let me connect until like 10 seconds have passed.

**Comment by jchristn** [comment]
Can you share some code that highlights this issue?

**Comment by MichaelEpicA** [comment]
Not really, all I have is some lua websocket code, and when I run it once, it works fine, if I run it twice, it fails. Along with once the client has been closed, it fails with connection failed, like it isn't even listening.

**Comment by jchristn** [comment]
Can you reproduce the behavior using the client and server test projects?

**Reply by MichaelEpicA** [reply]
I was able to reproduce the behavior, by changing the IP to * and the port to 8008, it fails to connect. Localhost works completely fine, but I need outside connections to work.

**Reply by MichaelEpicA** [reply]
I forgot to tell you that I did not have the IP as "localhost".

**Reply by jchristn** [reply]
Did you run as administrator or root?

**Reply by jchristn** [reply]
![image](https://github.com/jchristn/WatsonWebsocket/assets/1421081/8332e250-2385-4eeb-bcff-8185a0f5a45d)

**Reply by MichaelEpicA** [reply]
Yes, I did. Connections can come in fine, a connection comes in, it accepts it, once it disconnects, it fails for a bit until it reconnects.

**Reply by jchristn** [reply]
Can you reproduce with either the browser client code or the Test.Client project?

**Reply by MichaelEpicA** [reply]
Yes, I can.

**Reply by jchristn** [reply]
Could you provide the code you're using along with the steps?  Cheers

**Comment by jchristn** [comment]
I just tested using the ```Test.Server``` and ```Test.Client``` projects and could not reproduce this.  I'm not sure if the way I tested was the same as the way you did, but I:
- Started the server, left it running
- Started and connected a client
- Immediately terminated the process
- Started and connected another client
- Immediately terminated the process
- (repeat)

There was zero delay in establishing the new connection.
![image](https://github.com/jchristn/WatsonWebsocket/assets/1421081/c32716e9-77fe-46f0-ad9c-3f43fdbb8ee7)

I also tested by using a single client and continually disconnecting and reconnecting.  Again, no issue, no delay in establishing the new connection.
![image](https://github.com/jchristn/WatsonWebsocket/assets/1421081/f9731194-9971-4556-aa80-9afd20e93928)

**Comment by jchristn** [comment]
Moving to discussion, can't reproduce.
